### PR TITLE
Allow for expected kube-bench failures with bottlerocket

### PR DESCRIPTION
**Pull request by aidy**
Allow for expected kube-bench failures with bottlerocket
    
This should be fixed properly with https://github.com/aquasecurity/kube-bench/issues/808 and https://github.com/bottlerocket-os/bottlerocket/pull/1295
    
But, as these are known failures, this should not be a failure in our CI.
    
Workaround by patching kube-bench config for bottlerocket config paths, and allowing for a known number of failures.

connects #181

depends #178

### CODE REVIEW

File: test/cluster_test.go [hunk 1]
```diff
@@ -492,12 +497,280 @@ type KubeBenchResultTotals struct {
...
+          - "/etc/systemd/system/snap.kubelet.daemon.service"
+          - "/etc/systemd/system/snap.microk8s.daemon-kubelet.service"
+        bins:
+          - "hyperkube kubelet"
+          - "kubelet"
+        kubeconfig:
+          - "/etc/kubernetes/kubelet.conf"
+          - "/var/lib/kubelet/kubeconfig"
+          - "/etc/kubernetes/kubelet-kubeconfig"
+          - "/etc/kubernetes/kubelet/kubeconfig"
```
Comment (ettiee) [inline test/cluster_test.go:639]: Patched addition from [default config](https://github.com/aquasecurity/kube-bench/blob/main/cfg/config.yaml)

File: test/cluster_test.go [hunk 2]
```diff
@@ -492,12 +497,280 @@ type KubeBenchResultTotals struct {
...
+          - "/var/snap/microk8s/current/credentials/kubelet.config"
+        confs:
+          - "/var/lib/kubelet/config.yaml"
+          - "/var/lib/kubelet/config.yml"
+          - "/etc/kubernetes/kubelet/kubelet-config.json"
+          - "/etc/kubernetes/kubelet/config"
```
Comment (ettiee) [inline test/cluster_test.go:645]: Patched addition from [default config](https://github.com/aquasecurity/kube-bench/blob/main/cfg/config.yaml)

File: test/cluster_test.go [hunk 3]
```diff
@@ -492,12 +497,280 @@ type KubeBenchResultTotals struct {
...
+          - "proxy"
+        confs:
+          - /etc/kubernetes/proxy
+          - /etc/kubernetes/addons/kube-proxy-daemonset.yaml
+          - /etc/kubernetes/addons/kube-proxy-daemonset.yml
+          - /var/snap/kube-proxy/current/args
+          - /var/snap/microk8s/current/args/kube-proxy
+        kubeconfig:
+          - "/etc/kubernetes/kubelet-kubeconfig"
+          - "/etc/kubernetes/kubelet/config"
```
Comment (ettiee) [inline test/cluster_test.go:682]: Patched addition from [default config](https://github.com/aquasecurity/kube-bench/blob/main/cfg/config.yaml)

File: test/cluster_test.go [hunk 4]
```diff
@@ -492,12 +497,280 @@ type KubeBenchResultTotals struct {
...
+          - "kube-proxy"
+          - "hyperkube proxy"
+          - "hyperkube kube-proxy"
+          - "proxy"
+        confs:
+          - /etc/kubernetes/proxy
+          - /etc/kubernetes/addons/kube-proxy-daemonset.yaml
+          - /etc/kubernetes/addons/kube-proxy-daemonset.yml
+          - /var/snap/kube-proxy/current/args
+          - /var/snap/microk8s/current/args/kube-proxy
```
Comment (ettiee) [inline test/cluster_test.go:679]: Removed `/var/lib/kube-proxy/config.conf` from [default config](https://github.com/aquasecurity/kube-bench/blob/main/cfg/config.yaml#L163)

**Review by ettiee — COMMENTED** [review]
Note to self - check how the default config has been patched for kube-bench job

File: test/cluster_test.go (hunk 1 above)
Comment (ettiee) [reply test/cluster_test.go:639]: As per https://github.com/aquasecurity/kube-bench/pull/809/files

File: test/cluster_test.go (hunk 4 above)
Comment (ettiee) [reply test/cluster_test.go:679]: @aidy why was this line removed? (I'm working on the assumption that this patch is based off https://github.com/aquasecurity/kube-bench/pull/809/files)

**Review by ettiee — APPROVED** [review]
LGTM! I left a question out of curiosity having worked through the upstream fixes, but this looks 👌

File: test/cluster_test.go (hunk 4 above)
Comment (aidy) [reply test/cluster_test.go:679]: Not removed - but only added to kube-bench 5 hours ago! https://github.com/aquasecurity/kube-bench/pull/820

Comment (ettiee) [reply test/cluster_test.go:679]: Ah yes! 💡

**Review by dan-slinky-ckpd — APPROVED** [review]

**Pull request merged**